### ISSUE TITLE
uui-box: add a property to control header variant

### ISSUE THREAD

**Issue by iOvergaard** [umbraco/Umbraco.UI#447]
@bkclerke and @nathanwoulfe, should we perhaps consider adding a "header-variant" attribute to uui-box? I know that many UI libraries utilise this strategy to style a header a certain way while still providing a way to override the element used.

```
<uui-box header="Welcome" header-variant="h2"></uui-box>
```

_Originally posted by @iOvergaard in https://github.com/umbraco/Umbraco.CMS.Backoffice/issues/485#issuecomment-1480727845_

**Comment by Digipex** [comment]
Hey, I've been working on this, but ran into a hiccup and wondered if anyone could help.

In my forked repo I can't commit the changes I've made to then be able to submit a PR due to this error. 

`> git -c user.useConfigOnly=true commit --quiet --allow-empty-message --file -
[STARTED] Preparing lint-staged...
[SUCCESS] Preparing lint-staged...
[STARTED] Running tasks for staged files...
[STARTED] package.json — 6 files
[STARTED] *.ts — 5 files
[STARTED] *.{js,ts,mjs,cjs,html} — 5 files
[STARTED] * — 6 files
[STARTED] tsc-files --project tsconfig-lint-staged.json
[STARTED] eslint --cache --fix -c .eslintrc.js
[STARTED] prettier --ignore-unknown --write
[SUCCESS] prettier --ignore-unknown --write
[SUCCESS] * — 6 files
[FAILED] tsc-files --project tsconfig-lint-staged.json [FAILED]
[FAILED] tsc-files --project tsconfig-lint-staged.json [FAILED]
[FAILED] tsc-files --project tsconfig-lint-staged.json [FAILED]
[STARTED] Applying modifications from tasks...
[SKIPPED] Skipped because of errors from tasks.
[STARTED] Reverting to original state because of errors...
[FAILED] eslint --cache --fix -c .eslintrc.js [KILLED]
[FAILED] eslint --cache --fix -c .eslintrc.js [KILLED]
[SUCCESS] Reverting to original state because of errors...
[STARTED] Cleaning up temporary files...
[SUCCESS] Cleaning up temporary files...

✖ tsc-files --project tsconfig-lint-staged.json:
packages/uui-box/lib/uui-box.story.ts(5,20): error TS2307: Cannot find module '../README.md?raw' or its corresponding type declarations.

✖ eslint --cache --fix -c .eslintrc.js failed without output (KILLED).
husky - pre-commit hook exited with code 1 (error)`

I get this git error log. I've made changes to the README to mention the new attribute and also to the story to add the new usage.

Any ideas? 

Thanks

**Comment by iOvergaard** [comment]
Yeah sorry about that. Markdown changes don't pass the lint rules at the moment. You can bypass it by committing without the checks: `git commit --no-verify -m "message"`
